macOS/LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Views/TunnelControllerView/TunnelControllerViewModel.swift

```diff
@@ -291,6 +284,24 @@ public final class TunnelControllerViewModel: ObservableObject {
         }
     }
 
+    var isConnectedOrConnecting: Bool {
+        switch toggleTransition {
+        case .idle:
+            break
+        case .switchingOn:
+            return true
+        case .switchingOff:
+            return false
+        }
+
+        switch connectionStatus {
+        case .connecting, .connected, .reasserting:
+            return true
+        default:
+            return false
+        }
+    }
```

Comment:
This is useful for the VPN status badges.  Connected or connecting means protected, whereas anything else is a yellow badge (ie: unprotected).